packages/authorization-utils/src/domain.rs

```
@@ -11,54 +11,91 @@ pub enum Domain {
+                hyperlane_info.mailbox.clone()
+            }
         }
     }
 
-    pub fn get_callback_proxy_address(&self) -> Addr {
-        match &self.callback_proxy {
-            CallbackProxy::PolytoneProxy(address) => address.clone(),
+    pub fn get_callback_address(&self) -> Addr {
+        match &self.execution_environment {
+            ExecutionEnvironment::Cosmwasm(CosmwasmBridge::Polytone(polytone_info)) => {
+                polytone_info.polytone_proxy.clone()
+            }
+            ExecutionEnvironment::Evm(_, EvmBridge::Hyperlane(hyperlane_info)) => {
+                hyperlane_info.mailbox.clone()
+            }
         }
     }
 
-    pub fn get_connector_state(&self) -> PolytoneProxyState {
-        match &self.connector {
-            Connector::PolytoneNote { state, .. } => state.clone(),
+    pub fn get_polytone_proxy_state(&self) -> Option<PolytoneProxyState> {
+        match &self.execution_environment {
+            ExecutionEnvironment::Cosmwasm(CosmwasmBridge::Polytone(polytone_info)) => {
+                Some(polytone_info.polytone_note.state.clone())
+            }
+            ExecutionEnvironment::Evm(_, _) => None,
         }
     }
 
-    pub fn set_connector_state(&mut self, state: PolytoneProxyState) {
-        match &mut self.connector {
-            Connector::PolytoneNote {
-                state: current_state,
-                ..
-            } => {
-                *current_state = state;
+    pub fn set_polytone_proxy_state(
+        &mut self,
+        state: PolytoneProxyState,
+    ) -> Result<(), &'static str> {
+        match &mut self.execution_environment {
+            ExecutionEnvironment::Cosmwasm(CosmwasmBridge::Polytone(polytone_info)) => {
+                polytone_info.polytone_note.state = state;
+                Ok(())
             }
+            ExecutionEnvironment::Evm(_, _) => {
+                Err("EVM domain does not have a polytone proxy state")
+            }
```

Why not return an error here?

---

To make the question clearer, why not return a `StdError` or custom error here?

---

changed